### PR TITLE
CRM-16983 Dynamically load the content of Message templates in CiviMail when se…

### PR DESCRIPTION
**Pull request by seamuslee001**
…lected  rather than load on inital load

This loads the content of message templates on demand rather than on the initial load. I have tested this locally and seems like it works. Can someone else verify before merging

### CODE REVIEW

**Comment by davecivicrm** [comment]
Seamus - Please file a Jira (issue tracker) issue for this and clarify the problem(s) or bug(s) you're trying to solve.

**Comment by seamuslee001** [comment]
@davecivicrm have done so now https://issues.civicrm.org/jira/browse/CRM-16983

**Comment by davecivicrm** [comment]
Thanks @seamuslee001 - I've asked @totten to review.

**Comment by colemanw** [comment]
This looks like a good improvement, thanks @seamuslee001 .

File: ang/crmMailing/services.js [hunk 1]
```diff
@@ -60,17 +60,23 @@
...
-        id = '' + id; // parseInt(id);
         var dfr = $q.defer();
-        var tpl = _.where(tpls, {id: id});
-        if (id && tpl && tpl[0]) {
-          dfr.resolve(tpl[0]);
-        }
-        else {
-          dfr.reject(id);
-        }
+        var tpl = CRM.api3('MessageTemplate', 'get', {
```
Comment (totten) [inline ang/crmMailing/services.js]: From IRC discussion - this section could probably be simplified. This code is basically converting the jQuery-style promise (returned by CRM.api3) into an angular-style promise. The [`crmApi()`](https://github.com/civicrm/civicrm-core/blob/4.6/ang/crmUtil.js#L14) function will do that. If you call `crmApi('MessageTemplate','getsingle',...)`, then you can probably delete most of the `dfr` stuff.

**Comment by seamuslee001** [comment]
@totten Hey Tim, I have shifted to the Angular promise and can you have a look and test it. I think it is right 

Seamus

File: ang/crmMailing/services.js [hunk 2]
```diff
@@ -60,18 +60,15 @@
...
-        var dfr = $q.defer();
-        var tpl = _.where(tpls, {id: id});
-        if (id && tpl && tpl[0]) {
-          dfr.resolve(tpl[0]);
-        }
-        else {
-          dfr.reject(id);
-        }
-        return dfr.promise;
+        return crmApi('MessageTemplate', 'get', {
```
Comment (colemanw) [inline ang/crmMailing/services.js]: if you change "get" to "getsingle" then you can revert the changes to `ang/crmMailing/MsgTemplateCtrl.js` because it won't be buried in an array. Also you can then drop the "sequential" param.

**Comment by seamuslee001** [comment]
Thanks @colemanw have now made those changes, I think this should be fine if someone can just double check before merging that would be good
